x/gamm/keeper/gov/gov_test.go

```diff
@@ -0,0 +1,191 @@
+package gov_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/keeper/gov"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	balancertypes "github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+var (
+	defaultSwapFee    = sdk.MustNewDecFromStr("0.025")
+	defaultExitFee    = sdk.MustNewDecFromStr("0.025")
+	defaultPoolParams = balancer.PoolParams{
+		SwapFee: defaultSwapFee,
+		ExitFee: defaultExitFee,
+	}
+	defaultFutureGovernor = ""
+
+	// pool assets
+	defaultFooAsset = balancertypes.PoolAsset{
+		Weight: sdk.NewInt(100),
+		Token:  sdk.NewCoin("foo", sdk.NewInt(10000)),
+	}
+	defaultBarAsset = balancertypes.PoolAsset{
+		Weight: sdk.NewInt(100),
+		Token:  sdk.NewCoin("bar", sdk.NewInt(10000)),
+	}
+	defaultPoolAssets           = []balancertypes.PoolAsset{defaultFooAsset, defaultBarAsset}
+	defaultAcctFunds  sdk.Coins = sdk.NewCoins(
+		sdk.NewCoin("uosmo", sdk.NewInt(10000000000)),
+		sdk.NewCoin("foo", sdk.NewInt(10000000)),
+		sdk.NewCoin("bar", sdk.NewInt(10000000)),
+		sdk.NewCoin("baz", sdk.NewInt(10000000)),
+	)
+)
+
+func (suite *KeeperTestSuite) TestHandleSetSwapFeeProposal() {
+	tests := []struct {
+		fn func(poolId uint64)
+	}{
+		{
+			fn: func(poolId uint64) {
+				keeper := suite.app.GAMMKeeper
+
+				err := gov.HandleSetSwapFeeProposal(suite.ctx, *keeper, &types.SetSwapFeeProposal{
+					Title:       "tittle",
+					Description: "des",
+					Content: types.SetSwapFeeContent{
+						PoolId:  1,
+						SwapFee: sdk.MustNewDecFromStr("1.1"),
+					},
+				})
+
+				suite.Require().Error(err, types.ErrTooMuchSwapFee)
+			},
```

Comment:
Can we take these out and have them tested separately instead of testing them inside the function? Also, would it be possible to have more test cases?(superfluid module gov tests are a good example)

Reply:
Thank you for the comments. I'll fix them

Reply:
no problem! lmk once its ready for review!

Reply:
Yeah I think it's ready. Could u pls take a look